test/hotspot/jtreg/testlibrary_tests/template_framework/examples/TestWithTestFrameworkClass.java

```diff
@@ -0,0 +1,153 @@
+        //   - Data generated with Generators and hashtag replacement #con1.
+        // - GOLD value precomputed with dedicated call to test.
+        //   - This ensures that the GOLD value is computed in the interpreter
+        //     most likely, since the test method is not yet compiled.
+        //     This allows us later to compare to the results of the compiled
+        //     code.
+        //     The input data is cloned, so that the original INPUT_A is never
+        //     modified and can serve as identical input in later calls to test.
+        // - In the Setup method, we clone the input data, since the input data
+        //   could be modified inside the test method.
+        // - The test method makes use of hashtag replacements (#con2 and #op).
+        // - The Check method verifies the results of the test method with the
+        //   GOLD value.
+        var testTemplate = Template.make("op", (String op) -> body(
+            let("size", Generators.G.safeRestrict(Generators.G.ints(), 10_000, 20_000).next()),
+            let("con1", Generators.G.ints().next()),
+            let("con2", Generators.G.safeRestrict(Generators.G.ints(), 1, Integer.MAX_VALUE).next()),
+            """
+            // --- $test start ---
+            // $test with size=#size and op=#op
+            private static int[] $INPUT_A = new int[#size];
+            static {
+                Generators.G.fill(Generators.G.ints(), $INPUT_A);
+            }
+            private static int $INPUT_B = #con1;
+            private static Object $GOLD = $test($INPUT_A.clone(), $INPUT_B);
+
+            @Setup
+            public static Object[] $setup() {
+                // Must make sure to clone input arrays, if it is mutated in the test.
+                return new Object[] {$INPUT_A.clone(), $INPUT_B};
+            }
+
+            @Test
+            @Arguments(setup = "$setup")
+            public static Object $test(int[] a, int b) {
+                for (int i = 0; i < a.length; i++) {
+                    int con = #con2;
+                    a[i] = (a[i] * con) #op b;
+                }
+                return a;
+            }
+
+            @Check(test = "$test")
+            public static void $check(Object result) {
+                Verify.checkEQ(result, $GOLD);
+            }
+            // --- $test end   ---
+            """
+        ));
+
+        // Create a test for each operator..
+        List<String> ops = List.of("+", "-", "*", "&", "|");
+        List<TemplateToken> testTemplateTokens = ops.stream().map(testTemplate::asToken).toList();
+
+        // Create the test class, which runs all testTemplateTokens.
+        return TestFrameworkClass.render(
+            // package and class name.
+            "p.xyz", "InnerTest",
+            // List of imports. Duplicates are permitted.
+            List.of("compiler.lib.generators.*",
+                    "compiler.lib.ir_framework.*",
```

Comment:
```suggestion
```
This should not be needed since its imported by default in `TestFrameworkClass`. Or is this a deliberate duplication?